src/olympia/files/utils.py

```diff
@@ -830,6 +830,23 @@ def parse_xpi(xpi, addon=None, minimal=False):
     return check_xpi_info(xpi_info, addon, xpi)
 
 
+def validate_translations(translations):
+    if isinstance(translations['name'], dict):
+        for locale, localized_name in translations['name'].items():
+            if len(localized_name) > 50:
+                msg = ugettext(
+                    'Localized name (%s) should have fewer than 50 '
+                    'characters.')
+                raise forms.ValidationError(msg % locale)
+    if isinstance(translations['summary'], dict):
+        for locale, localized_summary in translations['summary'].items():
+            if len(localized_summary) > 500:
+                msg = ugettext(
+                    'Localized summary (%s) should have fewer than 500 '
+                    'characters.')
+                raise forms.ValidationError(msg % locale)
```

Comment:
I might be blind but where are these limits coming from?

Reply:
From here https://github.com/mozilla/addons-server/blob/0c879c8bd6d0968ac890b9876a0209db81b7c9fc/src/olympia/addons/forms.py#L446
And now I noticed it's for theme.  The value is different for extension. https://github.com/mozilla/addons-server/blob/0c879c8bd6d0968ac890b9876a0209db81b7c9fc/src/olympia/addons/forms.py#L140 . I don't know why..